2101. Detonate the Maximum Bombs/solution.py

```diff
@@ -0,0 +1,92 @@
+class Solution:
+    # Given a list of bombs, return the maximum number of bombs that
+    # will detonate as part of a chain reaction by just detonating one
+    # initial bomb. Each bomb is defined by a list of 3 
+    # ints representing the x and y coordinates of the bomb and the 
+    # radius affected by detonating the bomb. If the location of a
+    # bomb is inside the circular region of a bomb detonation, it
+    # will also detonate.
+    def maximumDetonation(self, bombs: List[List[int]]) -> int:
+        # Convert list of bombs into a directed graph where
+        # an edge exists from bomb1 to bomb2 if detonating bomb1
+        # would detonate bomb2 because the latter is within range
+        # of bomb1's blast.
+
+        # The graph is represented as a list of lists of bomb indices
+        # that is within range of the bomb and will be detonated by
+        # detonating the current bomb.
+        graph = []
+        for i in range(len(bombs)):
+            graph.append([])
+
+        # Iterate through every pair of bombs where order doesn't
+        # matter because we consider edges going both ways for a pair
+        # of bombs. So bomb1, bomb2 is the same as bomb2, bomb1
+        for i in range(len(bombs)):
+            (x1, y1, r1) = bombs[i]
+
+            for j in range(i + 1, len(bombs)):
+                (x2, y2, r2) = bombs[j]
+                
+                # For any two bombs, use the pythagorean theorem to
+                # determine the square of the distance between their
+                # locations.
+                distanceSquared = ((x1 - x2) * (x1 - x2)) + ((y1 - y2) * (y1 - y2))
```

Comment:
Hey Mommy you can just use (x1 - x2)**2. x**y means multiplying x by itself y times.